### PR TITLE
feat(esp_modem): Add optional ACT to operator-name

### PR DESCRIPTION
**Pull request by david-cermak**
Closes https://github.com/espressif/esp-protocols/issues/80

### CODE REVIEW

File: components/esp_modem/include/cxx_include/esp_modem_dce_module.hpp [hunk 1]
```diff
@@ -104,6 +104,15 @@ class GenericModule: public ModuleIf {
         pdp = std::move(new_pdp);
     }
 
+    /**
+     * @brief Simplified version of operator name (without the ACT, which is included in the command library)
+     */
+    command_result get_operator_name(std::string& name)
```
Comment (euripedesrocha) [inline components/esp_modem/include/cxx_include/esp_modem_dce_module.hpp:110]:
```suggestion
    command_result get_operator_name(std::string const& name)
```

File: components/esp_modem/include/cxx_include/esp_modem_dce_module.hpp [hunk 2]
```diff
@@ -104,6 +104,15 @@ class GenericModule: public ModuleIf {
...
+    {
+        int act;
+        return get_operator_name(name, act);
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply components/esp_modem/include/cxx_include/esp_modem_dce_module.hpp]: Renamed to `dummy_act`

File: components/esp_modem/include/cxx_include/esp_modem_dce_module.hpp (hunk 1 above)
Comment (david-cermak) [reply components/esp_modem/include/cxx_include/esp_modem_dce_module.hpp:110]: This couldn't be made const ref I'm afraid, the name parameter gets updated in the function.

File: components/esp_modem/src/esp_modem_command_library.cpp [hunk 3]
```diff
@@ -228,7 +228,7 @@ command_result set_flow_control(CommandableIf *t, int dce_flow, int dte_flow)
     return generic_command_common(t, "AT+IFC=" + std::to_string(dce_flow) + ", " + std::to_string(dte_flow) + "\r");
 }
 
-command_result get_operator_name(CommandableIf *t, std::string &operator_name)
+command_result get_operator_name(CommandableIf *t, std::string &operator_name, int &act)
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply components/esp_modem/src/esp_modem_command_library.cpp:231]: This part is largely generated from one source and all out parameters we pass by reference.

It's stated in the doxygen docs, that the param `act` is out:

https://github.com/espressif/esp-protocols/blob/a286634359593e9513f5a3383a0071598bcbefcb/components/esp_modem/include/generate/esp_modem_command_declare.inc#L50-L56

[the remark this answers is not in the document]
Comment (euripedesrocha) [reply components/esp_modem/src/esp_modem_command_library.cpp:231]: The usage  in `get_operator_name` still uses a reference to a local variable.

[the remark this answers is not in the document]
Comment (euripedesrocha) [reply components/esp_modem/src/esp_modem_command_library.cpp:231]: Forget last comment. It was a wrong reading of the code from my side.

**Pull request merged**